src/probe/probe.c

```diff
@@ -1099,23 +1183,69 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 		_probe->probe_points[first_free].stream_tag =
 			probe[i].stream_tag;
 
-		if (probe[i].purpose == PROBE_PURPOSE_LOGGING) {
+		if (fw_logs) {
 #if CONFIG_LOG_BACKEND_SOF_PROBE
 			probe_logging_init(probe_logging_hook);
 #else
 			return -EINVAL;
 #endif
 		} else {
-			notifier_register(_probe, dev->cb, NOTIFIER_ID_BUFFER_PRODUCE,
+#if CONFIG_IPC_MAJOR_4
+			notifier_register(&probe[i].buffer_id.full_id, buf, NOTIFIER_ID_BUFFER_PRODUCE,
 					  &probe_cb_produce, 0);
-			notifier_register(_probe, dev->cb, NOTIFIER_ID_BUFFER_FREE,
+			notifier_register(&probe[i].buffer_id.full_id, buf, NOTIFIER_ID_BUFFER_FREE,
```

Comment:
Oops, one compiler warning, can you fix this before merge:

 1196 |                         notifier_register(&probe[i].buffer_id.full_id, buf, NOTIFIER_ID_BUFFER_FREE,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1197 |                                           &probe_cb_free, 0);
      |                                           ~~~~~~~~~~~~~~~~~~